internal/store/hpa.go

```diff
@@ -76,6 +78,71 @@ var (
 				}
 			}),
 		},
+		{
+			Name: "kube_hpa_spec_target_metric",
+			Type: metric.Gauge,
+			Help: "The metric specifications used by this autoscaler when calculating the desired replica count.",
+			GenerateFunc: wrapHPAFunc(func(a *autoscaling.HorizontalPodAutoscaler) *metric.Family {
+				targets := make([]*metric.Metric, 0, len(a.Spec.Metrics))
+				for _, m := range a.Spec.Metrics {
+					var metricName string
+					var value, average int64
+					var valueOk, averageOk bool
+
+					switch m.Type {
+					case autoscaling.ObjectMetricSourceType:
+						metricName = m.Object.MetricName
+
+						value, valueOk = m.Object.TargetValue.AsInt64()
+						if m.Object.AverageValue != nil {
+							average, averageOk = m.Object.AverageValue.AsInt64()
+						}
+					case autoscaling.PodsMetricSourceType:
+						metricName = m.Pods.MetricName
+
+						average, averageOk = m.Pods.TargetAverageValue.AsInt64()
+					case autoscaling.ResourceMetricSourceType:
+						metricName = string(m.Resource.Name)
+
+						value = int64(*m.Resource.TargetAverageUtilization)
```

Comment:
Why aren't we nil checking here?

Reply:
Also, can we not use `.AsInt64()` here?

Reply:
I should definitely be checking `nil` here. 

`m.Resource.TargetAverageUtilization` is an `int32` so it just needs to be cast to an `int64`
https://github.com/kubernetes/api/blob/master/autoscaling/v2beta1/types.go#L173